### DISCUSSION TITLE
Configure a fixed IP wt32_eth01

### DISCUSSION THREAD

**Discussion by gonzalorui** [raomin/ESPAltherma#491] [Q&A]
Hi, someone help me configure a fixed IP over Ethernet? I've been trying for days without luck using the bgcngm code.

**Comment by bzq-pl** [comment]
I'm not sure if I done some additional changes in my version of code but in setup.h file I have added such likes:

```
#define WIFI_IP 10, 1, 0, 11
#define WIFI_SUBNET 255, 255, 255, 0
#define WIFI_GATEWAY 10, 1, 0, 1
#define WIFI_PRIMARY_DNS 10, 1, 0, 1     //optional
#define WIFI_SECONDARY_DNS 8, 8, 8, 8   //optional
```

maybe this will help you?

**Reply by gonzalorui** [reply]
Yes, but it doesn't work when I put the IP there. It seems that it is a configuration for wi-fi. It should be something like:

#define ETHERNET_IP 192, 168, 0, 254
#define ETHERNET_SUBNET 255, 255, 255, 0
#define ETHERNET_GATEWAY 192, 168, 0, 1
#define ETHERNET_PRIMARY_DNS 8, 8, 8, 8 // Optional
#define ETHERNET_SECONDARY_DNS 8, 8, 4, 4 // Optional

But it didn't work either. I don't quite understand how the code works.

**Comment by bzq-pl** [comment]
Sorry ... I give you wrong names of pre-processor variables :-(

In setup.h you have such section:

//Uncomment this to set a static IP instead of DHCP for the ESP (Separate by commas instead of dots)
//#define WIFI_IP 192, 168, 0, 5
//#define WIFI_SUBNET 255, 255, 255, 0
//#define WIFI_GATEWAY 192, 168, 0, 1
//#define WIFI_PRIMARY_DNS 8, 8, 8, 8     //A DNS address is needed, even if it's not used 
//#define WIFI_SECONDARY_DNS 8, 8, 4, 4   //A DNS address is needed, even if it's not used

and those variables are used in setup_wifi function:

void setup_wifi()
{
  delay(10);
  // We start by connecting to a WiFi network
  mqttSerial.printf("Connecting to %s\n", WIFI_SSID);

  #if defined(WIFI_IP) && defined(WIFI_GATEWAY) && defined(WIFI_SUBNET)
    IPAddress local_IP(WIFI_IP);
    IPAddress gateway(WIFI_GATEWAY);
    IPAddress subnet(WIFI_SUBNET);

    #ifdef WIFI_PRIMARY_DNS
      IPAddress primaryDNS(WIFI_PRIMARY_DNS);
    #else
      IPAddress primaryDNS();
    #endif

    #ifdef WIFI_SECONDARY_DNS
      IPAddress secondaryDNS(WIFI_SECONDARY_DNS);
    #else
      IPAddress secondaryDNS();
    #endif
...
}

just uncomment this section in setup.h and assign your correct IP definition.

**Reply by gonzalorui** [reply]
Sorry, I missed something. I put this code. But it doesn't get the correct IP.
--------------------------------------------------------------------------------------------------------------------------------------
//Default device hostname:
#define HOSTNAME "ESPAltherma"

//Setup your credentials and mqtt info here:
//only change the value between the " " leave the rest of the line untouched.
#define WIFI_SSID "SSID"//**Your SSID here**
#define WIFI_PWD "password"//**Your password here** leave empty if open (bad!)

//Uncomment this to set a static IP instead of DHCP for the ESP (Separate by commas instead of dots)
#define WIFI_IP 192, 168, 0, 254
#define WIFI_SUBNET 255, 255, 255, 0
#define WIFI_GATEWAY 192, 168, 0, 1
#define WIFI_PRIMARY_DNS 8, 8, 8, 8     //optional
#define WIFI_SECONDARY_DNS 8, 8, 4, 4   //optional

#define MQTT_SERVER "192.168.0.100"//**IP address here of your MQTT server**
#define MQTT_USERNAME ""//leave empty if not set (bad!)
#define MQTT_PASSWORD ""//leave empty if not set (bad!)
#define MQTT_PORT 1883
-----------------------------------------------------------------------------------------------------------
My board is a WT32-ETH01. It connects via ethernet and not via wi-fi.
Could you put the complete code (setup.h) for the case of ethernet connection with a fixed IP?###
